include/fiction/algorithms/graph/mincross.hpp

```diff
@@ -0,0 +1,624 @@
+        {
+            std::vector<uint64_t> penalty_array(fanout_ntk.rank_width(r + 1) + 1, 0);
+            uint64_t              max_pos = 0;
+
+            fanout_ntk.foreach_node_in_rank(r,
+                                            [this, &penalty_array, &max_pos](auto const& n)
+                                            {
+                                                std::vector<uint64_t> targets;
+
+                                                fanout_ntk.foreach_fanout(n,
+                                                                          [&](auto const& fanout)
+                                                                          {
+                                                                              uint64_t pos =
+                                                                                  fanout_ntk.rank_position(fanout);
+                                                                              targets.push_back(pos);
+                                                                          });
+
+                                                for (const auto pos : targets)
+                                                {
+                                                    for (auto k = pos + 1; k <= max_pos; ++k)
+                                                    {
+                                                        total_crossings += penalty_array[k];
+                                                    }
+                                                }
+
+                                                for (const auto pos : targets)
+                                                {
+                                                    max_pos = std::max(max_pos, pos);
+                                                    penalty_array[pos]++;
+                                                }
+                                            });
+        }
+    }
+
+    /**
+     * Logic network being reordered.
+     */
+    Ntk ntk;
+    /**
+     * Fanout-augmented view of the network for efficient traversal and connectivity queries.
+     */
+    mockturtle::fanout_view<Ntk> fanout_ntk;
+    /**
+     * Current total number of edge crossings in the network.
+     */
+    uint64_t total_crossings = std::numeric_limits<uint64_t>::max();
+    /**
+     * Stores median values used to sort nodes within ranks during optimization.
+     */
+    std::unordered_map<node, double> median_map;
+    /**
+     * Parameters for crossing minimization.
+     */
+    mincross_params ps;
+    /**
+     * Statistics that store the final number of crossings after optimization.
+     */
+    mincross_stats& pst;
+};
+
+/**
+ * Reimplementation of Graphviz's `mincross.c` algorithm for edge crossing minimization.
+ * This function reorders nodes in a leveled logic network to minimize the number of edge crossings using
+ * iterative median and transpose heuristics.
```

Comment:
Would be nice to have a link or other resource here that points to the original implementation from which this code is adapted.